### PR TITLE
[cmake][Issue #341 WAR] Move cuda_gen_all_arch to cudamapper

### PR DESCRIPTION
**Pull request by atadkase**
* Builds arch-specific binaries for cudamapper only.
* Builds SM30 PTX binaries for cudapoa and cudaaligner as a WAR for issue #341

### CODE REVIEW

**Comment by atadkase** [comment]
rerun tests

File: cudaaligner/CMakeLists.txt [hunk 1]
```diff
@@ -13,9 +13,11 @@ project(cudaaligner)
 # Project specific NVCC flags
 if((CUDA_VERSION_MAJOR GREATER 10) OR (CUDA_VERSION_MAJOR EQUAL 10 AND CUDA_VERSION_MINOR GREATER 0))
 set(CUDA_NVCC_FLAGS "${CUDA_NVCC_FLAGS} -std=c++14 --expt-relaxed-constexpr")
+
```
Comment (tijyojwad) [inline cudaaligner/CMakeLists.txt:16]: @atadkase can you remove this extra line here? I'm approving conditionally so once the tests have been run again after merge with master we can push it

Comment (atadkase) [reply cudaaligner/CMakeLists.txt:16]: Will do

Comment (atadkase) [reply cudaaligner/CMakeLists.txt]: Done.

**Pull request merged**